### PR TITLE
Add GHA for CI/CD that publishes the IdlImpTool as a global tool

### PR DESCRIPTION
**Pull request by jasonleenaylor**

### CODE REVIEW

File: .github/workflows/CI-CD.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      run: dotnet build
+
+    - name: Pack
+      run: dotnet pack IDLImporter --include-symbols --no-restore --no-build -o .
+
+    - name: Pack
+      run: dotnet pack IdlImpTool --include-symbols --no-restore --no-build -o .
+
+    - name: Publish to Nuget
+      run: dotnet nuget push *.nupkg -s https://api.nuget.org/v3/index.json -k ${{secrets.SILLSDEV_PUBLISH_NUGET_ORG}} --skip-duplicate
```
Comment (ermshiperete) [inline .github/workflows/CI-CD.yml:39]: I think we should also push (and publish below) the *.snupkg files

**Review by ermshiperete — APPROVED** [review]

File: .github/workflows/CI-CD.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - name: Restore
+      run: dotnet restore
+
+    - name: Build
+      run: dotnet build
+
+    - name: Pack
+      run: dotnet pack IDLImporter --include-symbols --no-restore --no-build -o .
+
+    - name: Pack
```
Comment (ermshiperete) [inline .github/workflows/CI-CD.yml:35]: We have `Pack` twice. Maybe name them `Pack IdlImporter` and `Pack IdlImpTool` ?

File: .github/workflows/CI-CD.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    - name: Publish to Nuget
+      run: dotnet nuget push *.nupkg -s https://api.nuget.org/v3/index.json -k ${{secrets.SILLSDEV_PUBLISH_NUGET_ORG}} --skip-duplicate
+      if: github.event_name == 'push' && matrix.os == 'ubuntu-latest'
+
+    - name: Publish Artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: NugetPackages
+        path: ./*.nupkg
+      if: github.event_name == 'pull_request'
```
Comment (ermshiperete) [inline .github/workflows/CI-CD.yml:47]: It's unclear (and probably random) whether the Windows or Ubuntu build gets published. Maybe we should make this more explicit?

```suggestion
      if: github.event_name == 'pull_request' && matrix.os == 'ubuntu-latest'
```

File: .github/workflows/CI-CD.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    - name: Install .NET Core
+      uses: actions/setup-dotnet@77a48bbb85cd3809790223e895ba4b6cbc110fe8
+      with:
+        dotnet-version: 6.0.x
+
+    - name: Restore
+      run: dotnet restore
+
+    - name: Build
+      run: dotnet build
```
Comment (ermshiperete) [inline .github/workflows/CI-CD.yml:30]: This includes a restore, so I think we can either remove the `Restore` step above, or pass `--no-restore` here.

File: .github/workflows/CI-CD.yml (hunk 1 above)
Comment (jasonleenaylor) [reply .github/workflows/CI-CD.yml:42]: As you can see in the log for the liblcm action the symbol packages get pushed as part of this command unless you give a flag to skip them.
https://github.com/sillsdev/liblcm/runs/7617839291?check_suite_focus=true
Are you sure we need them in the artifacts? I can add that if you promise to use them, I know I never would. 😄

**Pull request merged**